### PR TITLE
Harmonize CI and Gitian to use the same toolchain

### PR DESCRIPTION
**Pull request by patricklodder**
Given that ...

1. Ubuntu trusty (14.04) is fully end-of-life and there is no straightforward way to run it on GH Actions, but we use this for Gitian building
2. We have issues with the 1.14.x macOS SDK compiling on the latest Ubuntu LTS (focal, 20.04) and there has been no progress towards making that work (other coins use later SDKs, which we'll do with 1.21, to not breaking-change on a minor release)
3. We currently have the CI running everything on focal, except macOS, which runs on bionic (18.04).

... we have a discrepancy between errors getting caught on CI and what is getting caught when gitian building, sometimes leading to issues only being caught when doing a Gitian build, which is time and resource intensive.

This PR proposes harmonization between the GH Actions CI and Gitian by running all test suites and gitian on Ubuntu bionic which is supported with standard support until April 2023. This should give us the time we need to finish with Dogecoin Core 1.14 and have a stable 1.21 (or another major release), on which by then all new development should take place.

This PR:

1. Prepares the build process by:
    - Pulling in 2 upstream changes, 908c1d7 and 253f592 that enable gcc 7 / glib-2.27 builds
    - Mimicking fc6a9f2 from upstream on our 1.14 codebase
2. **_DOWNGRADES_** the GH Actions CI from focal to bionic wherever focal was used
3. Switches Gitian to build on bionic instead of trusty
4. Removes Trusty-specific patches by:
    - Pulling in the removal of a mingw patch from upstream that was sitting in the windows gitian descriptor: bd3f5a9
    - Removing the trusty-specific patch for zmq that we had to bring in with #2236

We will need some documentation changes and we can extend CI a bit to do some of the checks done in gitian (like symbol export checks) - I propose to pick those up separately to not make this PR too gigantic.

### CODE REVIEW

**Review by rnicoll — DISMISSED** [review]
Honestly taking a lot of these changes as "That looks fine even if I'm not sure why you're doing it", but utACK

**Comment by patricklodder** [comment]
> That looks fine even if I'm not sure why you're doing it

1. glibc compatibility is important to be able to run released Dogecoin binaries on older linux distros, where the glibc version is lower than the one used for compilation during the gitian process. This is fixed with 48d6330. Without it, changes between the minimum glibc version we support and the glibc version we compiled on will cause linkage to libc functions that had their signature changed, in particular `__divmoddi4` and `log2f` and this would cause issues when trying to run a gitian compiled binary with an older glibc on a system.
2. The bionic build also exports symbols for stdin, stdout and stderr and this was not allowed by `contrib/devtools/symbol-check.py`, even though this doesn't harm. 4ee7ff9 fixes this.
3. `in6addr_any` was getting exported which can be replaced by `IN6ADDR_ANY_INIT` - 45162cd

After these changes - which all come from upstream and are already part of `1.21-dev`, I just cherry-picked / reimplemented them - the build is not exporting any symbols that would cause incompatibility / overly depend on a particular glibc version. So b8a826f and 39c6df5 change all the builds to use bionic.

4. There are a number of patches introduced to just be compatible with trusty's build toolset: a problem with the trusty version of mingw that was 
    - preventing deterministic builds because the linker was leaking some bytes into a binary (ad4c630), and 
    - a patch we developed ourselves for zmq integration because we needed to include `ws2ipdef.h` on trusty for the gitian build of the libzmq dependency to work.

> utACK

As this touches the entire build system, I think that we really need testing of at least a couple of gitian builds to make sure that we still have a deterministic build, I don't feel comfortable merging this with an utACK alone...

**Comment by rnicoll** [comment]
Grabbing it for a Gitian build now

**Comment by rnicoll** [comment]
Gitian build done, will do a reindex with it overnight.

**Comment by rnicoll** [comment]
ACK:

Reindexed testnet up to block 644,630 without incident, so far. Going to sleep but will try to pick up the final result in the morning.

**Comment by rnicoll** [comment]
Testnet reindex completed without incident (using a Gitian built version)

**Comment by patricklodder** [comment]
My gitian hashes are:

```
720c51997183274720ab99f4782b78b8115746640272173ab981a474dc5bf9c6  dogecoin-1.14.5-aarch64-linux-gnu-debug.tar.gz
0e6cceee29591de26315ebecfea396a5c2c6773a94bfe96c1f3a9220bb303494  dogecoin-1.14.5-aarch64-linux-gnu.tar.gz
6c108d1e410d623013c4c5187b85bdcb7bdf5bf1d447f6878d567907eb9b11f9  dogecoin-1.14.5-arm-linux-gnueabihf-debug.tar.gz
e0e27b60736cac9c2a3730b634eff903fab6b5d552c9a2218fe60a5cad3a5395  dogecoin-1.14.5-arm-linux-gnueabihf.tar.gz
f343a4da5a55d8ba8eb602be64147c277372954773abe8a9a7ff9371951f7f37  dogecoin-1.14.5-i686-pc-linux-gnu-debug.tar.gz
07de58b5f11c63b2505f4d847cd927dd645a718742cf09e415507e1b76150954  dogecoin-1.14.5-i686-pc-linux-gnu.tar.gz
93ff826d8ccf016bb08e7dea1f336d325a0156292b4ee9efbe9368b63ea9b565  dogecoin-1.14.5-osx-unsigned.dmg
6cbec91159c89037cb44d82051260e7674749b2ca98bc061e34bfb30907f1c75  dogecoin-1.14.5-osx-unsigned.tar.gz
6105f9695d8525a1caac31cb317ef20b1af58d0b265160e8efb5b7622344b7d0  dogecoin-1.14.5-osx64.tar.gz
cd270fe882cdbde9540e3dd5f43e069f770d75e10ec9424472d72911cfabb24d  dogecoin-1.14.5-win32-debug.zip
41cfa1902a18d1784fe03e615a0d186e0322218c0cf347365b595bbe8c091c47  dogecoin-1.14.5-win32.zip
903533546ae2359f98ccce52fb51067fb6f6cf5722e292c0899bfc36453f7d82  dogecoin-1.14.5-win64-debug.zip
ba63801488513a25de1359a1280a61a4ac09642425165125659f9f2be5729650  dogecoin-1.14.5-win64.zip
57966d49ed41a6f30fc8926649f6d8a0b45afac94806db185804f069fdd57ae4  dogecoin-1.14.5-x86_64-linux-gnu-debug.tar.gz
5df51cc4e4d3ea294aa94c9e412d94c1f3cba1278c811dcf4d1114520e16cc86  dogecoin-1.14.5-x86_64-linux-gnu.tar.gz
```

@rnicoll could you please share yours?

**Comment by patricklodder** [comment]
Additional tests I've performed: tested macOS qt on testnet. It starts, syncs and [mined a block that was picked up by other 1.14.x and 1.10.0 nodes](https://chain.so/block/DOGETEST/45b11350ce2e5f8e1ee1afe898b9041ec9940743068d89114fe436e508fde4a5)

**Comment by rnicoll** [comment]
I've only got Linux to hand, doing OS X now, but mine differ:

```
b69211bb342119a71023710338b787b2ac76a5be46f04713c2915c4f602f7257 *dogecoin-1.14.5-aarch64-linux-gnu-debug.tar.gz
e925efdb5c084862e15ef7f404196de8f691b55e42c344c37242c97b0dc4247b *dogecoin-1.14.5-aarch64-linux-gnu.tar.gz
54efc5282e49042e24f263c1d4fde19f43d120b427d0e56846c71ed501917682 *dogecoin-1.14.5-arm-linux-gnueabihf-debug.tar.gz
7d04a4c0de6e0fa3c1af1234e8fac7c3b9791540f2069d3c05a94253390edbbc *dogecoin-1.14.5-arm-linux-gnueabihf.tar.gz
180d93b17617842882dbfcb9efeb6fd26d798fa96c91d2307af18699baaf9da7 *dogecoin-1.14.5-i686-pc-linux-gnu-debug.tar.gz
4b36230ef07e51cb2fa886ab70feeb77733c1508f16bd0a5834003955f83394c *dogecoin-1.14.5-i686-pc-linux-gnu.tar.gz
16e3b7683a380fde6ef9bc0ce36b92bfda8285357ac67925ee675907b0e12f03 *dogecoin-1.14.5-x86_64-linux-gnu-debug.tar.gz
badfa8335ac72e11d4482811afb590f39914d357f57a86ac6b9d69947ed2bdea *dogecoin-1.14.5-x86_64-linux-gnu.tar.gz
```

@langerhans @michilumin Can you do a build, moderate chance it's my Gitian (and if we can get 2/3 we'll know for certain). I've done Linux twice now, when I first got mismatched hashes I presumed I'd just mixed up files, but confirmed them.

I've used this as a tag https://github.com/rnicoll/dogecoin/tree/v1.14.5-bionic, let me know if that seems wrong.

**Comment by patricklodder** [comment]
> but mine differ

@rnicoll I get those same hashes when I build this tag with the trusty descriptors. You'll need to update the git repo where your descriptors are sitting to use that same tag.

**Comment by rnicoll** [comment]
@patricklodder Ah that's it, I thought I had already 🤦‍♂️ Rerunning now.

**Comment by rnicoll** [comment]
```
720c51997183274720ab99f4782b78b8115746640272173ab981a474dc5bf9c6  dogecoin-1.14.5-aarch64-linux-gnu-debug.tar.gz
0e6cceee29591de26315ebecfea396a5c2c6773a94bfe96c1f3a9220bb303494  dogecoin-1.14.5-aarch64-linux-gnu.tar.gz
6c108d1e410d623013c4c5187b85bdcb7bdf5bf1d447f6878d567907eb9b11f9  dogecoin-1.14.5-arm-linux-gnueabihf-debug.tar.gz
e0e27b60736cac9c2a3730b634eff903fab6b5d552c9a2218fe60a5cad3a5395  dogecoin-1.14.5-arm-linux-gnueabihf.tar.gz
8cdd7edc9db1d0e8924bb5e1906cd486a2c57ffae23f32b604f24e5db67a760f  dogecoin-1.14.5-i686-pc-linux-gnu-debug.tar.gz
7e2fcf8c9862a6baa7750f7bf1b8a75df6e5f3c1805cf5b51d6201a63218da69  dogecoin-1.14.5-i686-pc-linux-gnu.tar.gz
6105f9695d8525a1caac31cb317ef20b1af58d0b265160e8efb5b7622344b7d0  dogecoin-1.14.5-osx64.tar.gz
93ff826d8ccf016bb08e7dea1f336d325a0156292b4ee9efbe9368b63ea9b565  dogecoin-1.14.5-osx-unsigned.dmg
6cbec91159c89037cb44d82051260e7674749b2ca98bc061e34bfb30907f1c75  dogecoin-1.14.5-osx-unsigned.tar.gz
cd270fe882cdbde9540e3dd5f43e069f770d75e10ec9424472d72911cfabb24d  dogecoin-1.14.5-win32-debug.zip
60e3436f04680ce83430ca09ea419e7605f35b3977060bdc0d2e5ff3cea65452  dogecoin-1.14.5-win32-setup-unsigned.exe
41cfa1902a18d1784fe03e615a0d186e0322218c0cf347365b595bbe8c091c47  dogecoin-1.14.5-win32.zip
903533546ae2359f98ccce52fb51067fb6f6cf5722e292c0899bfc36453f7d82  dogecoin-1.14.5-win64-debug.zip
36588ec120d1ee1765b3c181cfe9d330c110c00390b57c7684d313ed09b76f99  dogecoin-1.14.5-win64-setup-unsigned.exe
ba63801488513a25de1359a1280a61a4ac09642425165125659f9f2be5729650  dogecoin-1.14.5-win64.zip
655c34538e1286e5af82164533889047f2005879ac1d362af92a0ab221592855  dogecoin-1.14.5-win-unsigned.tar.gz
b4dd734426d20ab0bcf7ec6deb34e035285e4db5bf95c6face1340e520562252  dogecoin-1.14.5-x86_64-linux-gnu-debug.tar.gz
06ee70a4c93dbd6b7b1d658ad014b5da09400b32ec9886084c46c6556e91eeeb  dogecoin-1.14.5-x86_64-linux-gnu.tar.gz
```

Closer. Probably I've screwed up something with Linux, re-running

**Comment by rnicoll** [comment]
Think I found the problem, didn't clear the Linux dependency build cache. Doing that now but won't have results until morning.

**Comment by rnicoll** [comment]
Linux rebuild gave me the same results. Can you diff the manifests please? https://pastebin.com/prDPKrmX

I'm on Gitian commit `cd8aa53c4e8870fa6be614a653024bae094c2f22`, ran `bin/make-base-vm --lxc --arch amd64 --suite bionic
` to create the VM, ` ./bin/gbuild --memory 3000 --commit dogecoin=v1.14.5-bionic ../dogecoin/contrib/gitian-descriptors/gitian-linux.yml -j6` to build.

**Comment by patricklodder** [comment]
here's the [diff.txt](https://github.com/dogecoin/dogecoin/files/7145288/diff.txt) (and my original [dogecoin-linux-1.14-res.yml.txt](https://github.com/dogecoin/dogecoin/files/7145462/dogecoin-linux-1.14-res.yml.txt))


`-` are mine that you don't have, `+` are yours that I don't have

something is odd with precompiled QT dependencies at the bottom:

```
-    2d4c2d03e160565fa84d0637e4996f5c65a01c4dca28cc6560e05c68257641f6  x86_64-linux-gnu/qt/qt-5.7.1-6bf56254403.tar.gz
-    ba64a9fa229a57354b196cbee51af0d0b767671a112d80e78864e97d2dc49e93  x86_64-linux-gnu/qt/qt-5.7.1-6bf56254403.tar.gz.hash
+    f3412e4364d379de12c339add088577a26e7a0a093dd7efbe450d389d94afb72  x86_64-linux-gnu/qt/qt-5.7.1-6bf56254403.tar.gz
+    e0801be213c64f19353b4c89391e35a3d71f09b5f470627aadcf59c86cf5f24e  x86_64-linux-gnu/qt/qt-5.7.1-6bf56254403.tar.gz.hash
```

which would explain why the packages with UI are different (i686 and x64_86) but not the ones without (aarch64 and armhf)

I cleaned my gitian-builder/cache directory before I switched from trusty to bionic but maybe there is something there - I'll do a build on a clean box and will report back.

**Comment by patricklodder** [comment]
After recompiling on a clean box, I still get `2d4c2d03e` for the compiled qt dependency and the tarball sha256sums match my previous ones.

Other differences besides a potential cache issue: My gitian builder is at `32b1145b0` and I use docker containers rather than lxc. I'll do an lxc build over the weekend to make sure.

**Comment by rnicoll** [comment]
I'll switch my builder over to `32b1145b0` too and see if that gets it.

**Comment by patricklodder** [comment]
Building with lxc I get the same hashes as you do on lxc. I'm investigating and reverting this back to draft in anticipation of a fix.

**Comment by patricklodder** [comment]
These are the differences between docker (`<`) and lxc (`>`) inside the compiled qt tarball:

```
< 4e5c45d621d670b733bdb6d10af96af68f8d8948be135a69bb157507e8bf81f5  ./plugins/platforminputcontexts/libcomposeplatforminputcontextplugin.a
---
> 74f077701e77be58808a99bf4cd8abe5651ff9787970a8d5b4a4c97a36c771f5  ./plugins/platforminputcontexts/libcomposeplatforminputcontextplugin.a
264c264
< d31db7934e340a1e0d15ceb1a27980086b577c4c6285795dba0c1f6a1abd6abc  ./lib/libQt5XcbQpa.a
---
> 15a93aece74f84e3cc91d0962c7227ed5234797c8c81f40ba77fbd6973fd795f  ./lib/libQt5XcbQpa.a
```

looks like some platform-specific information gets leaked compile-time with gcc7 but not with gcc4.8, I think that needs to get fixed. Will propose a patch.

Note: The fact that this was caught during gitian building underlines the importance of using different build environments for gitian. I'll propose something towards or on top of #2396 for that once we have this unblocked.

**Comment by patricklodder** [comment]
After doing a binary inspection, the problem with this is that within the qt depends build, both the QPA and the compose library are taking the system installed xcb library rather than the one provided and compiled with `depends`. 

Screen Shot 2021-09-12 at 23 32 44
Screen Shot 2021-09-12 at 23 34 25

This does not happen on the docker flavored build because gitian uses a docker hub image that does not contain any xkb traces, whereas the lxc flavored build uses `debootstrap`, which does include that. This causes the build to add a path to xkb and make some other compile-time choices based on the availability of these (and possibly other) x11 libs in `/usr/share`. So this is a build-time issue.

I'll try to patch this up on qt 5.7.1 but cannot guarantee success - we may need to upgrade the package.

**Comment by rnicoll** [comment]
If Docker avoids the bleed from system to container, I think the right answer is I retry with Docker and we move forward on that basis. Will try tonight/tomorrow.

**Comment by rnicoll** [comment]
Confirmed I get the same hashes with Docker (after clearing the cache).

Suggest we simply document that we do Gitian builds with Docker, and progress with this PR. Worst case one of the builders uses LXC by accident and ends up with a mis-matched build that's hard to diagnose. It's not the risk of something from the host leaking in that's a major concern, it's ensuring that even if one build host was compromised, it just wouldn't match the others.

**Comment by patricklodder** [comment]
> It's not the risk of something from the host leaking in that's a major concern, it's ensuring that even if one build host was compromised, it just wouldn't match the others.

What if the docker image is compromised?

**Comment by rnicoll** [comment]
> 
> 
> > It's not the risk of something from the host leaking in that's a major concern, it's ensuring that even if one build host was compromised, it just wouldn't match the others.
> 
> What if the docker image is compromised?

I mean the same applies for LXC or any other container, doesn't it? Or do you mean someone compromises the image on DockerHub (in which case, we have the same issue if someone compromises an Ubuntu image on the Ubuntu website/mirror sites)?

We're reliant on any attack on an origin source being identified and corrected quickly, so it's unlikely to affect multiple Gitian builders. Beyond that, any single Gitian builder could potentially be compromised (i.e. the host machine is hacked), but it's impractical to affect enough for the results to be consistent.

**Comment by patricklodder** [comment]
> I mean the same applies for LXC or any other container, doesn't it? Or do you mean someone compromises the image on DockerHub (in which case, we have the same issue if someone compromises an Ubuntu image on the Ubuntu website/mirror sites)?

Correct, but the DockerHub image and the image used by debootstrap for LXC are different and have different maintainers and different update schedules, and the image provider for vagrant is once more different. So with each different virtualization solution the detection capabilities grow.

> We're reliant on any attack on an origin source being identified and corrected quickly, so it's unlikely to affect multiple Gitian builders.

Within minutes, best case hours? Because that's often how fast the gitian building starts post-tag and we release quickly. We're often all building at the same time.


Bottom line: What's the benefit of not fixing a bug in determinism in a build system that is designed to be deterministic? What exactly are you advocating here?

**Comment by rnicoll** [comment]
I've presumed the Docker images are stable once the VM is created, i.e. it doesn't autoupdate from Dockerhub unless explicitely told to recreate the image. So someone would have to compromise it when we created the images, and the breach remain undetected until we used the images for live builds, wouldn't they?

**Comment by patricklodder** [comment]
That assumes you re-use your box with gitian-builder, don't purge your docker images, etcetera. With only 2 published builds / signatures on 1.14.4, yet there being a huge push for people to upgrade, in times where you yourself had to publish security warnings on twitter and reddit because of malware being promoted... I'd rather be safe than sorry.

That said, I've unblocked most of the other PRs now and have the others in the pipeline where I can, mostly running and gathering data, so I will look into this again now. Do expect to resolve this shortly.

**Comment by patricklodder** [comment]
UPDATE:

Found a [fix](https://github.com/patricklodder/dogecoin/commit/62bed9ab3957755426c3fe8e063387bee706019d) for the determinism issue. But then, I think I found the root cause for #2338 because the only way to make the build deterministic was by accidentally disabling xkb, which happens due to an error, which exists on both trusty and on bionic:

from my gitian (docker) linux build.log for OpenSSL 1.02:

```
31484:checking for xkbcommon... 
31485:make[1]: Entering directory `/home/ubuntu/build/dogecoin/depends/work/build/x86_64-linux-gnu/qt/5.7.1-5ba3dce7d75/qtbase/config.tests/unix/xkbcommon'
31486:g++ -c -pipe -pipe -O2 -I/home/ubuntu/build/dogecoin/depends/x86_64-linux-gnu/include -O2 -Wall -W -fPIC  -I. -I../../../mkspecs/linux-g++ -o xkbcommon.oxkbcommon.cpp
31487:xkbcommon.cpp:40:33: fatal error: xkbcommon/xkbcommon.h: No such file or directory
31488: #include <xkbcommon/xkbcommon.h>
31491:make[1]: *** [xkbcommon.o] Error 1
31492:make[1]: Leaving directory `/home/ubuntu/build/dogecoin/depends/work/build/x86_64-linux-gnu/qt/5.7.1-5ba3dce7d75/qtbase/config.tests/unix/xkbcommon'
31493:xkbcommon disabled.
```

from my gitian (lxc) linux build.log for https://github.com/patricklodder/dogecoin/commit/62bed9ab3957755426c3fe8e063387bee706019d:

```
33249:checking for xkbcommon... 
33250:make[1]: Entering directory '/home/ubuntu/build/dogecoin/depends/work/build/x86_64-linux-gnu/qt/5.7.1-960f1992275/qtbase/config.tests/unix/xkbcommon'
33251:g++ -c -pipe -pipe -O2 -I/home/ubuntu/build/dogecoin/depends/x86_64-linux-gnu/include -O2 -Wall -W -fPIC  -I. -I../../../mkspecs/linux-g++ -o xkbcommon.oxkbcommon.cpp
33252:xkbcommon.cpp:40:10: fatal error: xkbcommon/xkbcommon.h: No such file or directory
33253: #include <xkbcommon/xkbcommon.h>
33256:Makefile:183: recipe for target 'xkbcommon.o' failed
33257:make[1]: *** [xkbcommon.o] Error 1
33258:make[1]: Leaving directory '/home/ubuntu/build/dogecoin/depends/work/build/x86_64-linux-gnu/qt/5.7.1-960f1992275/qtbase/config.tests/unix/xkbcommon'
33259:xkbcommon disabled.
```

[Bitcoin compiles libxkbcommon since 0.22](https://github.com/bitcoin/bitcoin/commit/3272e34f9c548f7e4570d786fd322947e3cf20c6#diff-7f77230d4f9fb8f1952e63b8e5818c20ec62343614e6957c036b683ae28b7d80), so I'm looking into this in the following order:

1. backporting that without a QT update
2. 1, but then with a QT update

**Comment by patricklodder** [comment]
I checked rebasing this on top of #2595 at patricklodder@6ae2a6a1 and I no longer have a difference between lxc and docker with that. As soon as that is in, I'll rebase this and should be good.

**Comment by patricklodder** [comment]
These are my hashes for a8c035e, ran using a docker backend:

```
4040bd7e5a126af1f6cc158fec1468f8649ffdd1e07c201b56bc42902335fe3f  dogecoin-1.14.5-aarch64-linux-gnu-debug.tar.gz
16ed43ece509152ac1becb54665a71fa03ccf33ed54771c8449b5c888e55fa4e  dogecoin-1.14.5-aarch64-linux-gnu.tar.gz
85851e8183247aa236e681a6a3ff2bb89d50e3fb6bd76d0452e2327f9161bc39  dogecoin-1.14.5-arm-linux-gnueabihf-debug.tar.gz
5585d64af6458459b20ac271188150bfea240f3d424c90929e21dbbc6dcf58d5  dogecoin-1.14.5-arm-linux-gnueabihf.tar.gz
38b3170dcc16c1a3e68c1235ba14f8a61cda9a94b5bfd3269bafd389228937e6  dogecoin-1.14.5-i686-pc-linux-gnu-debug.tar.gz
1ff128cfb76cf6d44f9913cf46b982d89b2d89c9623e5b34d439c3047461c8f6  dogecoin-1.14.5-i686-pc-linux-gnu.tar.gz
12a63de6787034ff242e4df542e1e302af31d59cfbb05c038854c344e1920820  dogecoin-1.14.5-osx-unsigned.dmg
b80d1518f15d37cedeb463241462b9c1c3bb158ac88f00a53a5c135e102471bc  dogecoin-1.14.5-osx-unsigned.tar.gz
7cdcefaa5e538d16dcb18a37a42d34cf111d70745bafb463d051a6fa242b2506  dogecoin-1.14.5-osx64.tar.gz
c11f2df8e34dd36dac1136ee06fccb3c2a6263c8096f97e365a2a15b7a0cc741  dogecoin-1.14.5-win32-debug.zip
f7a24b88619b734f86d9713839a543861a294fb58922be66a94991225c0b407e  dogecoin-1.14.5-win32.zip
544a8aa4251ce1b1e1961ec9c16977264e770449f37d8d0943c1a0ea703d97c5  dogecoin-1.14.5-win64-debug.zip
0e1b1c197df24b4d3b2ca49c812ae1b2ad89ac2af8203572168beb21a5ed087b  dogecoin-1.14.5-win64.zip
bcf7ec746b170a1bde35a525dac13fc4ad66bbae8ec168bc3ee44f45ab73abca  dogecoin-1.14.5-x86_64-linux-gnu-debug.tar.gz
f4312cf51bb78a5fb228600f602db171a5a44225520a1cffb6f54a09245946f6  dogecoin-1.14.5-x86_64-linux-gnu.tar.gz
1bd639fad3e0fa1276822e6e5a23f25d19d2211c0a4a3ba302dbe7810cb2dd74  dogecoin-1.14.5.tar.gz
```

I'll compare those with lxc once more.

**Comment by patricklodder** [comment]
> I'll compare those with lxc once more.

Same output for me with lxc

**Comment by rnicoll** [comment]
Building the latest version now

**Comment by rnicoll** [comment]
Files all match, although I've got a few more I pulled out:

```
4040bd7e5a126af1f6cc158fec1468f8649ffdd1e07c201b56bc42902335fe3f *dogecoin-1.14.5-aarch64-linux-gnu-debug.tar.gz
16ed43ece509152ac1becb54665a71fa03ccf33ed54771c8449b5c888e55fa4e *dogecoin-1.14.5-aarch64-linux-gnu.tar.gz
85851e8183247aa236e681a6a3ff2bb89d50e3fb6bd76d0452e2327f9161bc39 *dogecoin-1.14.5-arm-linux-gnueabihf-debug.tar.gz
5585d64af6458459b20ac271188150bfea240f3d424c90929e21dbbc6dcf58d5 *dogecoin-1.14.5-arm-linux-gnueabihf.tar.gz
38b3170dcc16c1a3e68c1235ba14f8a61cda9a94b5bfd3269bafd389228937e6 *dogecoin-1.14.5-i686-pc-linux-gnu-debug.tar.gz
1ff128cfb76cf6d44f9913cf46b982d89b2d89c9623e5b34d439c3047461c8f6 *dogecoin-1.14.5-i686-pc-linux-gnu.tar.gz
12a63de6787034ff242e4df542e1e302af31d59cfbb05c038854c344e1920820 *dogecoin-1.14.5-osx-unsigned.dmg
b80d1518f15d37cedeb463241462b9c1c3bb158ac88f00a53a5c135e102471bc *dogecoin-1.14.5-osx-unsigned.tar.gz
7cdcefaa5e538d16dcb18a37a42d34cf111d70745bafb463d051a6fa242b2506 *dogecoin-1.14.5-osx64.tar.gz
f3bc126de165fb00a431138db4e1f86d2ee91cf35d8dff1a312121eb19fb4d9b *dogecoin-1.14.5-win-unsigned.tar.gz
c11f2df8e34dd36dac1136ee06fccb3c2a6263c8096f97e365a2a15b7a0cc741 *dogecoin-1.14.5-win32-debug.zip
24dd35848b6d703889d87b0ca6f23cef419f8264382dc5a1a175ce54ad731013 *dogecoin-1.14.5-win32-setup-unsigned.exe
f7a24b88619b734f86d9713839a543861a294fb58922be66a94991225c0b407e *dogecoin-1.14.5-win32.zip
544a8aa4251ce1b1e1961ec9c16977264e770449f37d8d0943c1a0ea703d97c5 *dogecoin-1.14.5-win64-debug.zip
5d4da6167cb1c4febc63d5e8821170688fda19ce2f66eee59c8850d082bd6a15 *dogecoin-1.14.5-win64-setup-unsigned.exe
0e1b1c197df24b4d3b2ca49c812ae1b2ad89ac2af8203572168beb21a5ed087b *dogecoin-1.14.5-win64.zip
bcf7ec746b170a1bde35a525dac13fc4ad66bbae8ec168bc3ee44f45ab73abca *dogecoin-1.14.5-x86_64-linux-gnu-debug.tar.gz
f4312cf51bb78a5fb228600f602db171a5a44225520a1cffb6f54a09245946f6 *dogecoin-1.14.5-x86_64-linux-gnu.tar.gz
```

Running full reindexes now just to be 100% sure

**Comment by rnicoll** [comment]
Testnet reindex complete. Mainnet up to late 2020 then turns out I didn't have the rest of the blocks downloaded on AWS, so that's taking a bit longer.

**Review by rnicoll — APPROVED** [review]
ACK, both on Linux & Windows
